pkg/controller/grafana/pluginsHelper.go

```diff
@@ -118,7 +104,8 @@ func (h *PluginsHelperImpl) FilterPlugins(cr *integreatly.Grafana, requested int
 		// Don't allow to install multiple versions of the same plugin
 		if filteredPlugins.HasSomeVersionOf(&plugin) == true {
 			installedVersion := filteredPlugins.GetInstalledVersionOf(&plugin)
-			h.AppendMessage(fmt.Sprintf("Not installing version %s of %s because %s is already installed", plugin.Version, plugin.Name, installedVersion.Version), plugin.Origin)
+			msg := fmt.Sprintf("Not installing version %s of %s because %s is already installed", plugin.Version, plugin.Name, installedVersion.Version)
+			common.KubeHelperImpl{}.AppendMessage(msg, plugin.Origin)
```

Comment:
This is creating a new instance of KubeHelper for every iteration of this line. Would be better to not make `AppendMessage` part of the `KubeHelperImpl` struct (by removing the (h KubeHelperImpl) part of the function definition).

Reply:
Or by using a shared instance of KubeHelper (could also be passed in or as a property of PluginHelper)